From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Juan Sanchez Fuentes <juanesf91@gmail.com>
Date: Fri, 29 Aug 2026 00:00:00 -0400
Subject: [PATCH 1/3] dt-bindings: pci: add Allwinner sun55i-a523 PCIe

Document the DWC glue for A523 (Cubie A5E) - 24MHz refclk, single
x1 lane, INNO combo-PHY, PL11 PERST# and vpcie3v3.

Signed-off-by: Juan Sanchez Fuentes <juanesf91@gmail.com>
Assisted-by: opencode:big-pickle
---
 Documentation/devicetree/bindings/pci/allwinner,sun55i-a523-pcie.yaml | 58 +++++++++++++++++++
 1 file changed, 58 insertions(+)

diff --git a/Documentation/devicetree/bindings/pci/allwinner,sun55i-a523-pcie.yaml b/Documentation/devicetree/bindings/pci/allwinner,sun55i-a523-pcie.yaml
new file mode 100644
index 000000000000..111111111111
--- /dev/null
+++ b/Documentation/devicetree/bindings/pci/allwinner,sun55i-a523-pcie.yaml
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/pci/allwinner,sun55i-a523-pcie.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Allwinner SUN55I A523 PCIe Host Controller (DWC glue)
+
+maintainers:
+  - Juan Sanchez Fuentes <juanesf91@gmail.com>
+
+description:
+  DWC-based PCIe RC for Allwinner SUN55I A523 (Radxa Cubie A5E). Glue
+  handles the INNO combo-PHY, slot regulators and PERST# as in the
+  downstream BSP, but uses the standard DWC host core and
+  msi_create_parent_irq_domain with a no-op bottom-chip ack for GICv3.
+
+properties:
+  compatible:
+    const: allwinner,sun55i-a523-pcie
+
+  reg:
+    maxItems: 1
+
+  reg-names:
+    const: dbi
+
+  clocks:
+    items:
+      - description: PCIe AUX clock
+      - description: 24MHz HOSC
+
+  clock-names:
+    items:
+      - const: pclk_aux
+      - const: hosc
+
+  resets:
+    maxItems: 1
+
+  phys:
+    maxItems: 1
+    description: INNO combo PHY in PCIe mode
+
+  phy-names:
+    const: pcie-phy
+
+  reset-gpios:
+    maxItems: 1
+    description: PERST#
+
+  wake-gpios:
+    maxItems: 1
+
+  num-lanes:
+    const: 1
+
+  vpcie3v3-supply:
+    description: 3.3V slot regulator (PL11 on Cubie A5E)
+
+required:
+  - compatible
+  - reg
+  - clocks
+  - clock-names
+  - phys
+  - phy-names
+
+additionalProperties: false
+
+examples:
+  - |
+    pcie@4800000 {
+        compatible = "allwinner,sun55i-a523-pcie";
+        reg = <0x04800000 0x480000>;
+        reg-names = "dbi";
+        clocks = <&osc24M>, <&ccu 135>;
+        clock-names = "hosc", "pclk_aux";
+        phys = <&combophy 1>;
+        phy-names = "pcie-phy";
+        reset-gpios = <&pio 7 11 1>;
+        num-lanes = <1>;
+        vpcie3v3-supply = <&reg_pcie_vcc3v3>;
+    };
--
2.39.5
